src/m4st/transcribe/transcribe.py

```diff
@@ -0,0 +1,49 @@
+from abc import ABC, abstractmethod
+from pathlib import Path
+
+import torch
+from transformers import AutoModelForSpeechSeq2Seq, AutoProcessor, pipeline
+
+
+class TranscriptionModel(ABC):
+    @abstractmethod
+    def __call__(self, path_to_file: Path) -> str:
+        r"""
+        Transcribe the speech from an audio file.
+        """
+
+
+class WhisperTranscription(TranscriptionModel):
+    def __init__(self, language: str):
+        self.device = "cuda:0" if torch.cuda.is_available() else "cpu"
+        self.language = language
+
+        torch_dtype = torch.float16 if torch.cuda.is_available() else torch.float32
+        model_id = "openai/whisper-large-v3"
```

Comment:
would be neater as an arg to `__init__`

Reply:
I did not mean this parameter to be modified, I generally design these classes to only work with a specific model. In fact, I think it might even be better to mark as 'private'